File: 2580-circular-sentence/circular-sentence.py
class Solution:
    def isCircularSentence(self, sentence: str) -> bool:
        firstChar = sentence[0]
        ans = True
        for i in range(len(sentence)):
            char = ""
            if i+1 >= len(sentence) or sentence[i+1] == " " :
                char = sentence[i]
                if i + 2 >= len(sentence):
                    ans =  char == firstChar
                else:
                    ans =  char == sentence[i + 2]
            if ans == False:
                return False
        
        return ans

File: 2580-circular-sentence/circular-sentence.js
/**
 * @param {string} sentence
 * @return {boolean}
 */
var isCircularSentence = function (sentence) {
    let firstChar = sentence[0]
    let ans = true
    for (let i = 0; i < sentence.length; i++) {
        let char = ""
        if (sentence[i + 1] == " " || sentence[i + 1] == undefined) {
            char = sentence[i]
        if (sentence[i + 2] == undefined) {
            ans = char == firstChar
        }
        else {
            ans = char == sentence[i + 2]
        }
        }
        if (ans == false) {
            return false
        }
    }
    return ans
};